java/java.lsp.server/vscode/src/panels/SshGuidePanel.ts

```diff
@@ -0,0 +1,132 @@
+    private static readonly statePropertyPrefix: StatePropertyPrefix = 'sshGuide';
+
+    private static readonly webviewsFolder: string = "webviews";
+    private static readonly scriptsFolder: string = "scripts";
+    private static readonly stylesFolder: string = "styles";
+
+    private readonly panel: vscode.WebviewPanel;
+    private disposables: vscode.Disposable[] = [];
+
+    public static createOrShow(context: vscode.ExtensionContext) {
+        if (SshGuidePanel.currentPanel) {
+            SshGuidePanel.currentPanel.panel.reveal();
+            return;
+        }
+
+        SshGuidePanel.currentPanel = new SshGuidePanel(context);
+    }
+
+    private constructor(context: vscode.ExtensionContext) {
+        this.panel = vscode.window.createWebviewPanel(
+            SshGuidePanel.viewType,
+            "Oracle Cloud Assets",
+            { viewColumn: vscode.ViewColumn.One, preserveFocus: true },
+            {
+                enableCommandUris: true,
+                enableScripts: true,
+                localResourceRoots: [vscode.Uri.file(path.join(context.extensionPath, SshGuidePanel.webviewsFolder))],
+            }
+        );
+        const iconPath: vscode.Uri = vscode.Uri.file(path.join(context.extensionPath, "images", "Apache_NetBeans_Logo.png"));
+        this.panel.iconPath = {
+            light: iconPath,
+            dark: iconPath,
+        };
+
+        this.setHtml(context);
+
+        this.panel.onDidDispose(
+            () => {
+                this.dispose();
+                removeGuidePropertiesFromStateFor(context, SshGuidePanel.statePropertyPrefix);
+            },
+            null,
+            this.disposables
+        );
+
+        this.panel.onDidChangeViewState(
+            () => {
+                if (this.panel.visible) {
+                    this.setHtml(context);
+                }
+            },
+            null,
+            this.disposables
+        );
+
+        this.panel.webview.onDidReceiveMessage(
+            (message) => {
+                if (message.command === "showSSHGuide") {
+                    const ocid: string = context.globalState.get(`${SshGuidePanel.statePropertyPrefix}-ocid`) || "";
+                    toggleGuideFor(SshGuidePanel.actionType, ocid);
+                }
+            },
+            undefined,
+            this.disposables
+        );
+    }
+
+    private async setHtml(context: vscode.ExtensionContext) {
+        let templateFile = path.resolve(__dirname, "../../webviews/ssh-guide.handlebars");
+        templateFile = await readFile(templateFile, "utf-8");
+        const template = Handlebars.compile(templateFile);
+        const ocid: string | undefined = context.globalState.get<string>(`${SshGuidePanel.statePropertyPrefix}-ocid`);
+
+        this.panel.webview.html = template({
+            cspSource: this.panel.webview.cspSource,
+            publicIp: context.globalState.get<string>(`${SshGuidePanel.statePropertyPrefix}-publicIp`),
```

Comment:
I don't recommend to use `context.globalState` for this. The globalState is a (persistent) memento, while this information seenms transient.

In addition, the caller currently sets a value into `globalState`, and then access **panel instance** (indirectly) through `SshGuidePanel.createOrShow` that either locates the current panel (an object instance) or creates one. So `createOrShow` could accept parameters, or Map or {} of parameters to pass down to the webview through the template context and store it in the `SshGuidePanel` instance then create the webview.